### PR TITLE
Uw 140 directory documentation

### PR DESCRIPTION
**Pull request by benjamin-cash**
**Description**

**Type of change**

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

**Checklist**

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation.  I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: docs/Contributors_Guide/directory_tree.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    test_utils
+      __init__.py
+      test_logger.py
+      test_errors.py
+    test_scheduler
+      __init__.py
+      test_slurm.py
+    test_runners
+      __init__.py
+      test_forecast.py
```
Comment (aerorahul) [inline docs/Contributors_Guide/directory_tree.rst:49]: What are runners?
Seems like out of scope of the workflow-tools that call out concrete use of a generic task.

File: docs/Contributors_Guide/directory_tree.rst [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  uwtools
+    __init__.py
+    scheduler
+      __init__.py
+      scheduler.py
+      slurm.py
+    config_parser
+      __init__.py
+      config_parser.py
```
Comment (aerorahul) [inline docs/Contributors_Guide/directory_tree.rst:58]: what is a config parser and how does it differentiate from a generic parser?

File: docs/Contributors_Guide/directory_tree.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  runners
+    __init__.py
+    forecast.py
+  helpers
+    __init__.py
+    logger.py
+    errors.py
```
Comment (aerorahul) [inline docs/Contributors_Guide/directory_tree.rst:65]: I think the correct one is `exceptions.py`

File: docs/Contributors_Guide/directory_tree.rst [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      __init__.py
+      scheduler.py
+      slurm.py
+    config_parser
+      __init__.py
+      config_parser.py
+  runners
+    __init__.py
+    forecast.py
+  helpers
```
Comment (aerorahul) [inline docs/Contributors_Guide/directory_tree.rst:62]: If `uwtools` are all helpers, then what stays in `uwtools`? 
Or is the idea that a directory cannot have files and directories?

File: docs/Contributors_Guide/directory_tree.rst [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+Python and follows this type of generic structure:
+
+Python structure::
+
+   pkg
+   ├── __init__.py
+   ├── module1.py
+   └── subpkg
+     ├── __init__.py
+     └── module2.py
```
Comment (aerorahul) [inline docs/Contributors_Guide/directory_tree.rst:23]: what is `pkg` and `subpkg`?  Are we going to allow installing just a `subpkg` from within a `pkg`?

**Review by aerorahul — COMMENTED** [review]
Some comments that could use some clarity.

**Comment by aerorahul** [comment]
I would like some clarity on this.  This repository is all helper tools, so what would land in `uwtools`? 
The inclusion of `runners` is not within the scope of this repository or set of tools.  Specifically, `uwtools` will not "run" a forecast per-se.
Also, these repos will "parse" files e.g. yaml.  And so, the tools will be something like a `yaml_parser`.  A config parser and validator belong to `workflow-configs` or something of the sort.

The tools in this repo are the ones in the red box on the right.

Screen Shot 2022-04-21 at 11 11 17 PM

File: docs/Contributors_Guide/directory_tree.rst (hunk 1 above)
Comment (benjamin-cash) [reply docs/Contributors_Guide/directory_tree.rst:49]: @christinaholtNOAA @fgabelmannjr  ?

Comment (benjamin-cash) [reply docs/Contributors_Guide/directory_tree.rst:49]: I think that this and the other files are meant to be illustrative, not prescriptive.

File: docs/Contributors_Guide/directory_tree.rst (hunk 3 above)
Comment (benjamin-cash) [reply docs/Contributors_Guide/directory_tree.rst]: Done.

File: docs/Contributors_Guide/directory_tree.rst (hunk 1 above)
Comment (fgabelmannjr) [reply docs/Contributors_Guide/directory_tree.rst:44]: I concur with @benjamin-cash .  My understanding of this task was to provide an illustrative example to contributors for how contributions should be organized within the repository rather than a prescriptive list of the structure.   However, as the framework matures, I could see this documentation becoming more prescriptive.

Comment (benjamin-cash) [reply docs/Contributors_Guide/directory_tree.rst:44]: I've made some edits to try and bring the illustrative/sample nature out a bit more.

**Comment by ope-adejumo** [comment]
Based on the changes so far, This PR looks good to me.

**Review by ope-adejumo — APPROVED** [review]
PR looks good and ready.
